The build system tries to link to installed system libraries first, which 
fails because the ABI has changed...

Index: mlt-0.9.2/src/swig/ruby/build
===================================================================
--- build       2015-01-29 17:15:55.914492487 +0100
+++ build1      2015-01-29 17:16:12.234493276 +0100
@@ -5,8 +5,10 @@
   exit 0
 end
 system( "ln -sf ../mlt.i" )
+system( "ln -sf ../../framework/libmlt.so" )
+system( "ln -sf ../../mlt++/libmlt++.so" )
 system( "swig -c++ -ruby -I../../mlt++ -I../.. mlt.i" )
 $CFLAGS += " -I../.. " + (ENV.has_key?('CXXFLAGS')? ENV['CXXFLAGS'] : '')
-$LDFLAGS += " -L../../mlt++ -lmlt++"
+$LIBS += " -lmlt++ -lmlt -lstdc++"
 create_makefile('mlt')
 system( "make" )

